flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStore.java

```diff
@@ -213,14 +214,26 @@ public RetrievableStateHandle<T> addAndLock(String key, T state)
 
         // initialize flag to serve the failure case
         boolean discardState = true;
+        final AtomicInteger retryNum = new AtomicInteger(0);
         try {
             // a successful operation will result in the state not being discarded
             discardState =
                     !updateConfigMap(
                                     cm -> {
+                                        retryNum.incrementAndGet();
                                         try {
                                             return addEntry(cm, key, serializedStoreHandle);
                                         } catch (Exception e) {
+                                            // It could happen the fabric8 k8s client retries a
+                                            // transaction that has already succeeded due to network
+                                            // issues. We let the AlreadyExistException caused by
+                                            // PossibleInconsistentStateException here to avoid
+                                            // discarding the state.
+                                            if (retryNum.get() > 1
+                                                    && e instanceof AlreadyExistException) {
+                                                e.initCause(
```

Comment:
I'm wondering whether we actually want to expose the `PossiblyInconsistentStateException` here. This would trigger a warning on [CheckpointCoordinator:1413](https://github.com/apache/flink/blob/88b309b7dcad269ad084eab5e2944724daf6dee4/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java#L1413). I'd assume that this warning is not needed in this specific case. The `addEntry` method worked as expected. No error needs to be exposed and the related Checkpoint will be handled as normal.

Reply:
I agree with you that the WARNING log in the `CheckpointCoordinator` is unnecessary. But how could this cause the checkpoint creation for the current run to fail? It is actually triggered successfully without subsuming the old one. And it will be subsumed in the next triggering checkpoint.

Reply:
If you are suggesting to simply break the retry since the `addEntry` executed successfully instead of throwing the `PossiblyInconsistentStateException`, I could also update the current implementation.

Reply:
> I agree with you that the WARNING log in the CheckpointCoordinator is unexpected. But how could this cause the checkpoint creation for the current run to fail? It is actually triggered successfully without subsuming the old one. And it will be subsumed in the next triggering checkpoint.

The warning we talked about is created in the `catch` block in [CheckpointCoordinator:1411-1431](https://github.com/apache/flink/blob/88b309b7dcad269ad084eab5e2944724daf6dee4/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java#L1411-L1431). At the end of the `catch` block, a `CheckpointException` is thrown which will cause the checkpoint to fail as far as I can see.

Reply:
So you prefer to not expose any errors and let the current checkpoint handle as normal. Right?